api/src/nlp/controllers/nlp-entity.controller.ts

```diff
@@ -222,6 +222,21 @@ export class NlpEntityController extends BaseController<
     if (!ids?.length) {
       throw new BadRequestException('No IDs provided for deletion.');
     }
+
+    const { count: builtinNlpEntitiesCount } = await this.filterCount({
+      _id: { $in: ids },
+      builtin: true,
+    });
+
+    if (builtinNlpEntitiesCount > 0) {
+      this.logger.warn(
+        `Unable to delete NLP entities with provided IDs: ${ids}`,
+      );
+      throw new MethodNotAllowedException(
+        'Deletion failed: Selection includes built-in NLP entities that are protected',
+      );
+    }
```

Comment:
Okay for the frontend side, but for the API, I suggest that we simply add this condition to the criteria at the base repository level. WDYT ?

{ ...criteria,  builtin: { $ne: true } }